packages/meditrak-app/app/database/types/Option.js

```diff
@@ -11,8 +11,9 @@ export class Option extends RealmObject {
       id: this.id,
       value: this.value,
       label: this.label,
-      sortOrder: this.sortOrder,
+      sort_order: this.sortOrder,
       attributes: this.attributes ? JSON.parse(this.attributes) : {},
+      option_set_id: this.optionSetId,
```

Comment:
I had a try with inverse relationship in Realm and seems like the linkingObjects property is _always going to be a collection_ containing all objects linked to that one (so maybe it's only meant for many to many relationships?). However, the relationship between Option and OptionSet is one to many (1 option only belongs to 1 option set). So to make it work, we have to add this to properties:
```
optionSets: {
      type: 'linkingObjects',
      objectType: 'OptionSet',
      property: 'options',
    },
```

And then maybe have a separate getter method to get our option set:
```
getOptionSet() {
    return this.optionSets[0];
  }
```

It seems that's the only work around. What do you reckon about this? I'm a bit afraid that it might cause some confusion when someone tries to access the Option's optionSet property. I guess we can also add some documentation?

Reply:
True! So we'd need to set up the primary link to be the other way around, so that `option.optionSet` was directly linked, and then `optionSets.options` used a `linkingObjects` list.

Can you spend 30 minutes looking at how hard it would be to migrate the primary link to be the other way around? If it's complex, just leave it as you have it here - it's pretty straightforward.

Reply:
I did also have a try with the other way around too, but it didn't work. The schema I had was:
```
Option: {
	optionSet: 'OptionSet'
}
```
and

```
OptionSet: {
	options: {
		type: 'linkingObjects',
		objectType: 'Option',
		property: 'optionSet',
    },
}
```

The reason it didn't work was, in our case, we want the Option.optionSet to be automatically updated/linked whenever we add a new option into OptionSet.options. But with the schema above, it's the other way around, the list of options will be automatically updated when we set optionSet in an option (because options are the `linkingObjects`). It also requires to change the logic of adding options into option set. So that's why I kinda gave up... Hope you're fine with that.

Reply:
Have set it up! It's working fine now!

Reply:
Amazing! Thank you @chris-bes, I appreciate you putting in the extra effort